src/main/java/emissary/output/DropOffUtil.java

```diff
@@ -1018,15 +1018,23 @@ public void processMetadata(final List<IBaseDataObject> payloadList) {
             // save specified metadata items for children to grab
             parentTypes.put("" + level, p.getFileType());
 
-            final String fn = p.getStringParameter("Original-Filename");
-            if (fn != null && fn.indexOf(".") > -1) {
-                final int pos = fn.lastIndexOf(".") + 1;
-                if (pos < fn.length()) {
-                    final String fext = fn.substring(pos);
-                    if (fext.length() > 0 && fext.length() <= this.maxFilextLen) {
-                        p.setParameter("FILEXT", fext.toLowerCase());
+            if (p.hasParameter("Original-Filename")) {
+                final List<String> extensions = new ArrayList<>();
```

Comment:
Recommend extracting lines 1022:1034 to a new method.

Reply:
What's the rationale for the recommendation? Is this duplicated elsewhere? Where you looking for an associated test for this block of code?

Reply:
Agreed that this method extraction would make the `processMetadata` method more coherent, but I don't consider that a blocker for the PR.  We see virtually the same pattern immediately after this FILEXT processing (https://github.com/NationalSecurityAgency/emissary/pull/160/files#diff-5e7393b5cf28de1fa844df12051fecba96c6962dcbf4aaa3be6b7c233677e410R1040-R1060 would also benefit from a similar method extraction).

Reply:
Just concerned about the overall length and complexity of ```processMetadata```. I believe extracting to method will make it  more readable.

Reply:
Addressed in 78aa26094eeec7db567017c574de866820e6c0a0